Build is failing 

```
         D:\agent\1\s\src\Compatibility\ControlGallery\src\Core\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\GalleryPages_TitleView.xaml.sg.cs(11,44): error CS0122: 'MemberNotNullAttribute' is inaccessible due to its protection level [D:\agent\1\s\src\Compatibility\ControlGallery\src\Core\Compatibility.ControlGallery.Core-net6.csproj]
```

--- a/src/Controls/src/SourceGen/CodeBehindGenerator.cs
+++ b/src/Controls/src/SourceGen/CodeBehindGenerator.cs
@@ -168,10 +168,16 @@ namespace Microsoft.Maui.Controls.SourceGen
 
 				// add MemberNotNull attributes
 				if (namedFields != null)
+				{
+					sb.AppendLine($"#if NET5_0_OR_GREATER");
 					foreach ((var fname, _, _) in namedFields)
 					{
+
 						sb.AppendLine($"\t\t[global::System.Diagnostics.CodeAnalysis.MemberNotNullAttribute(nameof({(CSharpKeywords.Contains(fname) ? "@" + fname : fname)}))]");
 					}
+
+					sb.AppendLine($"#endif");
+				}
 
 				sb.AppendLine("\t\tprivate void InitializeComponent()");
 				sb.AppendLine("\t\t{");

